tests/test_engine.py

```diff
@@ -65,6 +65,15 @@ def flow_for_tests(dog: str, cat: int):
     return flow_for_tests
 
 
+@pytest.fixture
+def parameterized_flow_with_default_arguments():
+    @flow
+    def flow_for_tests(foo: str = "bar", bar: int = 1):
+        """Flow for testing functions"""
```

Comment:
What do you think about confirming that the defaults are used?

```suggestion
        """Flow for testing functions"""
        
        assert foo == "bar"
        assert bar == 1
```

Reply:
Hmmm what are we looking to capture with this test?
I'm wondering if it's better to define this flow in `test_does_not_raise_signature_mismatch_on_missing_default_args` and then assert defaults were used based on the function output (i.e. return `foo` and `bar` and check their vals in `.result()`)

Reply:
Just updated, how's that?

Reply:
Ah yes, that's even better, awesome!